src/librawspeed/decoders/ArwDecoder.cpp

```diff
@@ -352,9 +357,7 @@ void ArwDecoder::DecodeLJpeg(const TiffIFD* raw) {
 
   PostProcessLJpeg();
 
-  const TiffEntry* size_entry = raw->hasEntry(TiffTag::SONYRAWIMAGESIZE)
-                                    ? raw->getEntry(TiffTag::SONYRAWIMAGESIZE)
-                                    : raw->getEntry(TiffTag::DEFAULTCROPSIZE);
+  const TiffEntry* size_entry = raw->getEntry(TiffTag::SONYRAWIMAGESIZE);
```

Comment:
so no fallback is necessary, we're always sure to have this EXIF marker available?

Reply:
> we're always sure to have this EXIF marker available?

Seems to be the case so far.

In the worst case, I'd rather return the uncropped image w/ black borders than the DNG crop, I feel it's more consistent with the stated behaviour:

> RawSpeed does NOT…
> 
> crop the image to the same sizes as manufactures, but supplies biggest possible images.

Reply:
FWIW, `getEntry()` throws if you query for an entry that will not be found.